src/benchmarklib/benchmark_sql_executor.cpp

```diff
@@ -0,0 +1,86 @@
+#include "benchmark_sql_executor.hpp"
+
+#include "concurrency/transaction_manager.hpp"
+#include "logical_query_plan/jit_aware_lqp_translator.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "utils/check_table_equal.hpp"
+#include "utils/timer.hpp"
+#include "visualization/lqp_visualizer.hpp"
+#include "visualization/pqp_visualizer.hpp"
+
+namespace opossum {
+BenchmarkSQLExecutor::BenchmarkSQLExecutor(bool enable_jit, const std::shared_ptr<SQLiteWrapper>& sqlite_wrapper,
+                                           const std::optional<std::string>& visualize_prefix)
+    : _enable_jit(enable_jit),
+      _sqlite_wrapper(sqlite_wrapper),
+      _visualize_prefix(visualize_prefix),
+      _transaction_context(TransactionManager::get().new_transaction_context()) {}
+
+std::shared_ptr<const Table> BenchmarkSQLExecutor::execute(const std::string& sql) {
+  auto pipeline_builder = SQLPipelineBuilder{sql};
+  if (_visualize_prefix) pipeline_builder.dont_cleanup_temporaries();
+  pipeline_builder.with_transaction_context(_transaction_context);
+  if (_enable_jit) pipeline_builder.with_lqp_translator(std::make_shared<JitAwareLQPTranslator>());
+
+  auto pipeline = pipeline_builder.create_pipeline();
+
+  const auto& result_table = pipeline.get_result_table();
+  metrics.emplace_back(std::move(pipeline.metrics()));
+
+  if (_sqlite_wrapper) {
+    _verify_with_sqlite(pipeline);
+  }
+
+  if (_visualize_prefix) {
+    _visualize(pipeline);
+  }
+
+  return result_table;
+}
+
+void BenchmarkSQLExecutor::_verify_with_sqlite(SQLPipeline& pipeline) {
+  const auto sqlite_result = _sqlite_wrapper->execute_query(pipeline.get_sql());
+  const auto& result_table = pipeline.get_result_table();
```

Comment:
why not `get_result_tables()`? the way it is done now, only the last statement in a query if verified against sqlite?

Reply:
As far as I know, the SQLite wrapper cannot handle multiple statements with multiple results yet. I'd rather require the pipeline to contain only a single statement.

Reply:
> I'd rather require the pipeline to contain only a single statement

I'm okay with that. Is there an Assert enforcing this?

Reply:
Soon.